packages/shared/src/store/groupActions.ts

```diff
@@ -609,3 +609,186 @@ export async function deleteNavSection(group: db.Group, navSectionId: string) {
     });
   }
 }
+
+export async function kickUserFromGroup({
+  groupId,
+  contactId,
+}: {
+  groupId: string;
+  contactId: string;
+}) {
+  logger.log('kicking user from group', groupId, contactId);
+
+  const existingGroup = await db.getGroup({ id: groupId });
+
+  if (!existingGroup) {
+    console.error('Group not found', groupId);
+    return;
+  }
+
+  if (!existingGroup.members) {
+    console.error('Group members not found', groupId);
+    return;
+  }
+
+  if (!existingGroup.members.find((member) => member.contactId === contactId)) {
+    console.error('User not found in group', groupId, contactId);
+    return;
+  }
+  // optimistic update
+  await db.removeChatMembers({
+    chatId: groupId,
+    contactIds: [contactId],
+  });
+
+  try {
+    await api.kickUsersFromGroup({
+      groupId,
+      contactIds: [contactId],
+    });
+  } catch (e) {
+    console.error('Failed to kick user from group', e);
+    // rollback optimistic update
+    await db.addChatMembers({
+      chatId: groupId,
+      type: 'group',
+      contactIds: [contactId],
+    });
+  }
+}
+
+export async function getBannedUsers(groupId: string) {
```

Comment:
unused? If we want this, seems like it should be a db hook?

Reply:
I'll just remove